samples/constitutions/default/actions.js

```diff
@@ -762,6 +762,25 @@ const actions = new Map([
       }
     ),
   ],
+  [
+    "set_js_engine_options",
+    new Action(
+      function (args) {
+        checkType(args.max_heap_size, "integer", "max_heap_size");
+        checkType(args.max_stack_size, "integer", "max_stack_size");
```

Comment:
A convention we've adopted in the configuration is to suffix the unit name, eg. _days, _count. We do use _size in places where we allow strings and interpret the suffix instead, eg. max_msg_size: "4MB".

Here I would suggest suggest either max_heap_bytes, int, or max_heap_size, string, allowing "4MB" etc.